### PR TITLE
Add authors to rever

### PR DESCRIPTION
**Pull request by CJ-Wright**

### CODE REVIEW

File: .authors.yml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+[{name: Christopher J. Wright, email: cjwright4242gh@gmail.com, num_commits: 3, first_commit: 2017-07-25
```
Comment (scopatz) [inline .authors.yml]: Hmm this should have nicer YAML formatting.  Not sure why it came out like this...

Comment (scopatz) [reply .authors.yml]: Also you need to add "christopher" to your aliases list to get the mailmap to generate correctly.

File: .mailmap [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+#
+#   good-name <good-email> bad-name <bad-email>
+#
+# You can skip bad-name if it is the same as good-name and is unique in the repo.
+#
+# This file is up-to-date if the command git log --format="%aN <%aE>" | sort -u
+# gives no duplicates.
+
+Anthony Scopatz <scopatz@gmail.com>
+christopher <cjwright4242@gmail.com>
```
Comment (scopatz) [inline .mailmap]: This line presumably should be different

File: AUTHORS [hunk 3]
```diff
@@ -0,0 +1,13 @@
+All of the people who have made at least one contribution to rever.
+Authors are sorted by number of commits.
+
+* Anthony Scopatz
+* christopher
```
Comment (scopatz) [inline AUTHORS]: And this line shouldn't be here :)

**Comment by CJ-Wright** [comment]
I think we might need a regenerate authors command. That way when changes are made to the authors yaml we can update the generated files. The setup command created a thing with Christopher. I then changed the yaml file but I don't know how to regenerate the other files and make sure things are correct.

**Comment by scopatz** [comment]
`rever -f -a authors setup` will regenerate the authors

**Comment by CJ-Wright** [comment]
I got:
```python
(regro) christopher@christopher-ubuntu ~/dev/rever rever_authors $ rever -f -a authors setup
Activity 'authors' has already been setup!
```

**Comment by scopatz** [comment]
That seems like a bug.

**Comment by scopatz** [comment]
Looking into it

**Comment by scopatz** [comment]
Fix is in #166

**Comment by scopatz** [comment]
Ok merge from master and try it again :)

**Comment by CJ-Wright** [comment]
Ok, I'll add docs on the force setup for doing this again

**Comment by scopatz** [comment]
Thanks!

**Comment by CJ-Wright** [comment]
```python
Re-doing acctivity 'authors' which has already been setup!
Authors file 'AUTHORS' exists AND Mailmap file '.mailmap' exists
rever forced, overwriting files!
'igor.ghisi@gmail.com' does not have 'Igor T. Ghisi' listed! Please add the name to 'igor.ghisi@gmail.com' in '.authors.yml'. For example:

- name: Igor T. Ghisi
  email: igor.ghisi@gmail.com

Or add an alias:

- name: Some Body Else
  email: igor.ghisi@gmail.com
  aliases:
    - Igor T. Ghisi
Traceback (most recent call last):
  File "/home/christopher/mc/envs/regro/bin/rever", line 6, in <module>
    exec(compile(open(__file__).read(), __file__, 'exec'))
  File "/home/christopher/dev/rever/scripts/rever", line 3, in <module>
    main()
  File "/home/christopher/dev/rever/rever/main.xsh", line 233, in main
    env_main(args=args)
  File "/home/christopher/dev/rever/rever/main.xsh", line 225, in env_main
    setup_activities(ns)
  File "/home/christopher/dev/rever/rever/main.xsh", line 179, in setup_activities
    status = act.setup()
  File "/home/christopher/dev/rever/rever/activity.xsh", line 103, in setup
    status = self._setup()
  File "/home/christopher/dev/rever/rever/activities/authors.xsh", line 170, in setup_func
    md = self._update_authors(filename, template, format, metadata, sortby)
  File "/home/christopher/dev/rever/rever/activities/authors.xsh", line 178, in _update_authors
    md = update_metadata(metadata)
  File "/home/christopher/dev/rever/rever/authors.xsh", line 113, in update_metadata
    _verify_names_emails_aliases(y, by_names, by_emails, filename)
  File "/home/christopher/dev/rever/rever/authors.xsh", line 89, in _verify_names_emails_aliases
    raise RuntimeError("Duplicated author/email combos")
RuntimeError: Duplicated author/email combos
```

**Comment by scopatz** [comment]
Yeah that seems like it is supposed to happen.  As the error message says, you need to edit the `.authors.yml` file

**Comment by scopatz** [comment]
I don't still don't understand how you got such a weird `.authors.yml` file.  You may want to blow it away and start again.

**Comment by scopatz** [comment]
My `.authors.yml` file looks like:

```yaml
- name: Anthony Scopatz
  email: scopatz@gmail.com
  num_commits: 215
  first_commit: 2017-05-16 14:52:07
- name: Marius van Niekerk
  email: marius.v.niekerk@gmail.com
  num_commits: 0
  first_commit: 2018-10-13 11:39:05
- name: Dougal J. Sutherland
  email: dougal@gmail.com
  num_commits: 1
  first_commit: 2018-08-11 13:07:29
- name: Christopher J. Wright
  email: cjwright4242gh@gmail.com
  num_commits: 3
  first_commit: 2017-07-25 20:41:13
- name: Igor T. Ghisi
  email: igor.ghisi@gmail.com
  num_commits: 4
  first_commit: 2018-02-26 18:09:46
- name: Aaron Meurer
  email: asmeurer@gmail.com
  num_commits: 34
  first_commit: 2017-06-23 14:44:39
- name: Christoph Deil
  email: Deil.Christoph@gmail.com
  num_commits: 1
  first_commit: 2018-10-17 16:35:00
- name: christopher
  email: cjwright4242@gmail.com
  num_commits: 114
  first_commit: 2017-08-31 14:00:05
```

**Comment by CJ-Wright** [comment]
I think there is a bug in this, I'm going to open a new PR but the duplicate stuff is not doing the right thing.

**Pull request closed without merging**